src/devices/Lis3Dh/Lis3Dh.cs

```diff
@@ -0,0 +1,187 @@
+            this[Register.CTRL_REG2] = 0b00000000;
+
+            // interrupts settings to default
+            this[Register.CTRL_REG3] = 0b00000000;
+
+            // CTRL_REG4 is handled by ChangeSettings (AccelerationScale and high-resolution mode bit)
+
+            // reboot, FIFO, interrupt settings to default
+            this[Register.CTRL_REG5] = 0b00000000;
+
+            // other boot and interrupt settings to default
+            this[Register.CTRL_REG6] = 0b00000000;
+        }
+
+        private void ChangeSettings(DataRate dataRate, OperatingMode operatingMode, AccelerationScale accelerationScale)
+        {
+            byte dataRateBits = (byte)((byte)dataRate << 4);
+            byte lowPowerModeBitAndAxesEnable = (byte)(operatingMode == OperatingMode.LowPowerMode ? 0b1111 : 0b0111);
+            this[Register.CTRL_REG1] = (byte)(dataRateBits | lowPowerModeBitAndAxesEnable);
+
+            // remainder of the bits (block data update/endianess/self-test/SPI 3 or 4-wire setting) set to default
+            byte fullScaleBits = (byte)((byte)accelerationScale << 4);
+            byte highResolutionModeBit = (byte)(operatingMode == OperatingMode.HighResolutionMode ? 0b1000 : 0b0000);
+            this[Register.CTRL_REG4] = (byte)(fullScaleBits | highResolutionModeBit);
+
+            _dataRate = dataRate;
+            _operatingMode = operatingMode;
+            _accelerationScale = accelerationScale;
+        }
+
+        private Lis3Dh Initialize(DataRate dataRate, OperatingMode operatingMode, AccelerationScale accelerationScale)
+        {
+            ResetUnusedSettings();
+            ChangeSettings(dataRate, operatingMode, accelerationScale);
+
+            // return this to simplify syntax
+            return this;
+        }
+
+        private Vector3 ReadRawAcceleration()
+        {
+            Span<byte> vec = stackalloc byte[6];
+            ReadRegister(Register.OUT_X_L, vec);
+
+            short x = BinaryPrimitives.ReadInt16LittleEndian(vec.Slice(0, 2));
+            short y = BinaryPrimitives.ReadInt16LittleEndian(vec.Slice(2, 2));
+            short z = BinaryPrimitives.ReadInt16LittleEndian(vec.Slice(4, 2));
+            return new Vector3(x, y, z);
+        }
+
+        private float GetAccelerationDivisor() => _accelerationScale switch
+        {
+            // Max is power of 2 so we don't lose any data here by integer division
+            AccelerationScale.Scale02G => Max / 2,
+            AccelerationScale.Scale04G => Max / 4,
+            AccelerationScale.Scale08G => Max / 8,
+
+            // Note: While we theoretically should divide by 16 here,
+            //       according to my measurements the actual divisor is around 24:
+            //       When 24 is used measurements match other scales and 1G is closer to 1G rather than 0.64G.
+            //       I was not able to find section in datasheet explaining why this is 24 (exactly 50% more).
+            AccelerationScale.Scale16G => Max / 24,
```

Comment:
v-beer for anyone who can find something about in the [spec](https://www.st.com/resource/en/datasheet/lis3dh.pdf)